pkg/kthena-router/router/router.go

```diff
@@ -255,44 +255,99 @@ func (r *Router) HandlerFunc() gin.HandlerFunc {
 
 func (r *Router) doLoadbalance(c *gin.Context, modelRequest ModelRequest) {
 	modelName := modelRequest["model"].(string)
-	// step 3: Find pods and model server details
-	// Get gateway key from context if available (set by Gateway listener)
-	var gatewayKey string
-	if key, exists := c.Get(GatewayKey); exists {
-		if k, ok := key.(string); ok {
-			gatewayKey = k
+
+	// Check if this is an InferencePool request from HTTPRoute
+	var pods []*datastore.PodInfo
+	var port int32
+	var modelServerName types.NamespacedName
+	var modelRoute *v1alpha1.ModelRoute
+	var modelServer *v1alpha1.ModelServer
+
+	if inferencePoolName, exists := c.Get("inferencePoolName"); exists {
```

Comment:
I am thinking a more graceful solution should be convert inferencePool to a synthetic  modelSever, so we would only need to handle modelServer in our data path